app/models/AllGainsSessionModel.scala

```diff
@@ -0,0 +1,105 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.gains._
+import play.api.libs.json.{Json, OFormat}
+import uk.gov.hmrc.crypto.EncryptedValue
+
+case class AllGainsSessionModel(allGains: Seq[PolicyCyaModel], gateway: Option[Boolean] = None) {
+  def toSubmissionModel: GainsSubmissionModel = {
+
+    def convertEmptyToOption[A](items: Seq[A]): Option[Seq[A]] = if (items.isEmpty) None else Some(items)
+
+    val life: Option[Seq[LifeInsuranceModel]] = convertEmptyToOption(allGains.filter(elem => elem.policyType.contains("Life Insurance")).map(cya =>
+      LifeInsuranceModel(
+        customerReference = cya.policyNumber,
+        event = cya.policyEvent,
+        gainAmount = cya.amountOfGain.get,
+        taxPaid = cya.treatedAsTaxPaid,
+        yearsHeld = cya.yearsPolicyHeld,
+        yearsHeldSinceLastGain = cya.yearsPolicyHeldPrevious,
+        deficiencyRelief = cya.deficiencyReliefAmount)
+    ))
+    val capital: Option[Seq[CapitalRedemptionModel]] = convertEmptyToOption(allGains.filter(elem => elem.policyType.contains("Capital Redemption")).map(cya =>
+      CapitalRedemptionModel(
+        customerReference = cya.policyNumber,
+        event = cya.policyEvent,
+        gainAmount = cya.amountOfGain.get,
+        taxPaid = cya.treatedAsTaxPaid,
+        yearsHeld = cya.yearsPolicyHeld,
+        yearsHeldSinceLastGain = cya.yearsPolicyHeldPrevious,
+        deficiencyRelief = cya.deficiencyReliefAmount)
+    ))
+    val lifeAnnuity: Option[Seq[LifeAnnuityModel]] = convertEmptyToOption(allGains.filter(elem => elem.policyType.contains("Life Annuity")).map(cya =>
+      LifeAnnuityModel(
+        customerReference = cya.policyNumber,
+        event = cya.policyEvent,
+        gainAmount = cya.amountOfGain.get,
+        taxPaid = cya.treatedAsTaxPaid,
+        yearsHeld = cya.yearsPolicyHeld,
+        yearsHeldSinceLastGain = cya.yearsPolicyHeldPrevious,
+        deficiencyRelief = cya.deficiencyReliefAmount)
+    ))
+
+    val voidedIsa: Option[Seq[VoidedIsaModel]] = convertEmptyToOption(allGains.filter(elem => elem.policyType.contains("Voided ISA")).map(cya =>
+      VoidedIsaModel(
+        customerReference = cya.policyNumber,
+        event = cya.policyEvent,
+        gainAmount = cya.amountOfGain.get,
+        taxPaidAmount = cya.taxPaidAmount,
+        yearsHeld = cya.yearsPolicyHeld,
+        yearsHeldSinceLastGain = cya.yearsPolicyHeldPrevious
+      ))
+    )
+
+    val foreign: Option[Seq[ForeignModel]] = convertEmptyToOption(allGains.filter(elem => elem.policyType.contains("Foreign Policy")).map(cya =>
+      ForeignModel(
+        customerReference = cya.policyNumber,
+        gainAmount = cya.amountOfGain.get,
```

Comment:
Should we make https://github.com/hmrc/income-tax-additional-information/pull/24/files#diff-56cfd51e869f05b033d1d3b0523edb6834f86f77ae8626afc929f6495c5b9446R25 an Option to avoid using .get here?

Reply:
After a discussing this with Rob, and was advised that `gainAmount` will always need a value provided, so rather than change the attribute to be an option, I could provide a zero value to it - `gainAmount = cya.amountOfGain.getOrElse(BigDecimal(0))`